### PR TITLE
feat(semantic-release): implement an automated changelog

### PR DESCRIPTION
**Pull request by n-rodriguez**

### CODE REVIEW

File: docs/README.rst [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+
+``kitchen test``
+^^^^^^^^^^^^^^^^
+
+Runs all of the stages above in one go: i.e. ``destroy`` + ``converge`` + ``verify`` + ``destroy``.
+
+``kitchen login``
+^^^^^^^^^^^^^^^^^
+
+Gives you SSH access to the instance for manual testing.
```
Comment (myii) [inline docs/README.rst]: The use of heading symbols in this file needs to be reviewed because the `Testing` section is ending up as a sub-section.

Comment (myii) [reply docs/README.rst]: While making this change, could you pull in the `.gitignore` as well?

File: kitchen.yml [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+        - apt-get update && apt-get install -y udev locales && rm -rf /var/lib/apt/lists/*
+        - localedef -i en_US -c -f UTF-8 -A /usr/share/locale/locale.alias en_US.UTF-8
+        - echo "en_US.UTF-8 UTF-8" >> /etc/locale.gen
+        - locale-gen en_US.UTF-8
+      platform: ubuntu
+
+provisioner:
+  name: salt_solo
+  log_level: info
+  salt_version: latest
```
Comment (myii) [inline kitchen.yml]: Would it be possible to get the pre-salted images?  Or is that for a subsequent PR?

**Comment by myii** [comment]
@n-rodriguez Are we good to finish this off?

File: kitchen.yml (hunk 2 above)
Comment (n-rodriguez) [reply kitchen.yml]: I'll do it in a future PR

File: docs/README.rst [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+
+``syslog_ng.config``
+^^^^^^^^^^^^^^^^^^^^
+This state manages the file ``syslog_ng.conf`` under ``/etc/syslog-ng`` (template found in "syslog_ng/files"). The configuration is populated by values in "syslog_ng/map.jinja" based on the package's default values (and RedHat, Debian, Suse and Arch family distribution specific values), which can then be overridden by values of the same name in pillar.
+
+Testing
+-------
+
+Linux testing is done with ``kitchen-salt``.
+
```
Comment (myii) [inline docs/README.rst]: Can we get the `Requirements` sub-section here, as merged in https://github.com/saltstack-formulas/template-formula/pull/123?  Of course, if you have anything you want to improve here, please do so.

File: .gitignore [hunk 4]
```diff
@@ -1,2 +1,110 @@
...
+/site
+
+# mypy
+.mypy_cache/
+
+# Bundler
 Gemfile.lock
+
+# copied `.md` files used for conversion to `.rst` using `m2r`
+docs/*.md
```
Comment (myii) [inline .gitignore]: There's been another change since in `template-formula`!

```suggestion
docs/*.md

# Vim
*.sw?
```

**Comment by myii** [comment]
@n-rodriguez Merged this, there's been plenty of effort already.  I'll submit the remaining minor changes in another PR.

**Pull request merged**